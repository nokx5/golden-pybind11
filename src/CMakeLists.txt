# add_library(bitwise golden.cpp)

# add_subdirectory(./pybind_mask)
add_subdirectory(./pybind_view)
