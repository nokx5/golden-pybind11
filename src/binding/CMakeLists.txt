# BUILD
pybind11_add_module(pyview pybind_view/wrapper_view.cpp)

# # BUILD WITHOUT CMAKE PYBIND11 MACROS message(STATUS "fancy pybind linking")
# find_package(PythonInterp 3) find_package(PythonLibs 3) add_library(pyview
# MODULE pybind_view/wrapper_view.cpp) target_link_libraries(pyview
# ${PYTHON_LIBRARIES}) set_target_properties(pyview PROPERTIES OUTPUT_NAME
# "pyview" PREFIX "") target_include_directories(pyview PRIVATE
# ${PYTHON_INCLUDE_DIRS}) # BUILD WITHOUT CMAKE PYBIND11 MACROS

# * INSTALLATION
install(TARGETS pyview DESTINATION bin)
