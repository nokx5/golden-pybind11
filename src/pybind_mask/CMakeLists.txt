# pybind11_add_module(pymask wrapper_mask.cpp) target_link_libraries(np_view
# dynamic_builder)

# install( TARGETS pymask CONFIGURATIONS Release RUNTIME DESTINATION bin)

# get_target_properties(pymask)
