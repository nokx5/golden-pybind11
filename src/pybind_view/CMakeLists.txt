# * BUILD
pybind11_add_module(pyview wrapper_view.cpp)
# message(STATUS "fancy pybind linking") find_package(PythonInterp 3)
# find_package(PythonLibs   3) add_library(pyview MODULE wrapper_view.cpp)
# target_link_libraries(pyview ${PYTHON_LIBRARIES}) set_target_properties(pyview
# PROPERTIES OUTPUT_NAME "pyview" PREFIX "") target_include_directories(pyview
# PRIVATE ${PYTHON_INCLUDE_DIRS})

# * INSTALLATION

install(TARGETS pyview DESTINATION bin)
