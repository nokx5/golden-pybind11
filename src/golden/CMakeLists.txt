set(libraries gold)

foreach(library ${libraries})

  if(${PROJECT_STATIC})
    add_library(${library} STATIC core/${library}.cpp)
  else()
    add_library(${library} SHARED core/${library}.cpp)
  endif()
  target_link_libraries(${library}) # ${BOOST_LIBRARIES} Boost::system
                                    # Boost::filesystem

endforeach()

set(clicommands golden)

foreach(clicommand ${clicommands})

  add_executable(cli_${clicommand} cli/${clicommand}.cpp)
  # target_link_libraries(cli_${clicommand} ${BOOST_LIBRARIES})

  install(
    TARGETS cli_${clicommand}
    CONFIGURATIONS Release
    DESTINATION bin)

endforeach()
